src/record/record-transition.js

```diff
@@ -157,13 +155,12 @@ RecordTransition.prototype.add = function (socketWrapper, version, message) {
       socketWrapper.sendError(C.TOPIC.RECORD, C.EVENT.INVALID_MESSAGE_DATA, message.raw)
       return
     }
-    if (message.data.length === 5) {
-      try {
-        this._applyConfig(update, message.data[4])
-      } catch (e) {
-        update.sender.sendError(C.TOPIC.RECORD, 'INVALID_CONFIG_DATA', config)
-        return
-      }
+
+    try {
+      this._applyConfig(update, message)
```

Comment:
you could move the `update.sender` into the applyconfig method and that way also send the error message

invalid_config_data should also be an event constant and should be record specific

`INVALID_RECORD_CONFIG_DATA`

ideally you would also send the record name so that the front end can figure out what went wrong. Either just the record name or the raw message